Sources/SwiftSyntax/SyntaxOtherNodes.swift

```diff
@@ -16,16 +16,11 @@
 public struct UnknownSyntax: SyntaxProtocol, SyntaxHashable {
   public let _syntaxNode: Syntax
 
-  /// Convert the given `Syntax` node to an `UnknownSyntax` if possible. Return
-  /// `nil` if the conversion is not possible.
-  public init?(_ syntax: Syntax) {
-    guard syntax.raw.kind == .unknown else { return nil }
-    self._syntaxNode = syntax
+  public init?<S: SyntaxProtocol>(_ syntaxNode: S) {
+    guard syntaxNode.raw.kind == .unknown else { return nil }
+    self._syntaxNode = Syntax(syntaxNode)
   }
 
-  /// Creates an `UnknownSyntax` node from the given `SyntaxData`. This assumes
-  /// that the `SyntaxData` is of the correct kind. If it is not, the behaviour
-  /// is undefined.
```

Comment:
Is this intentional? I'm not against this change, but could you do this for other types syntax nodes too?

Reply:
I was removing comments where the protocol already had them since they seemed mostly pointless in that case. This one is unintentional though because this is just defined by each node. If we want to get rid of it we could define an internal conformance I guess, but I'll leave it for now.